massgov/mayflower
# Update mf/react to work with shared assets

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/dev/docs/change-log-instructions.md)

## Description
Updates mf/react to use the shared assets in the project root.
During builds, assets are copied into src/assets so that they can be bundled in /es/assets, then they are cleaned up so that they only exist at /assets to avoid confusion.

The UMD build (unused) is removed, it was not working with path aliases

When `npm start` is run, a symbolic link is created instead because cleanup cannot be done reliably. This also obviates the need for a special assets-watch update during local dev.

## Related Issue / Ticket

- [JIRA issue]()
- [Github issue]()

## Steps to Test


1. check out code, run all `npm run` commands
1. everything should work as expected:
** npm run build should result in a /es/ build where paths resolve to /es/assets
** npm start should work because of the /src/assets symbolic link
** SharedAssets alias will work during npm start and storybook builds because of the resolver in react/.storybook/webpack.config.js 
** npm lint will work because of the path alias resolver in react/.eslintrc.json
** in case src/assets is accidentally left in place during dev, it is git ignored so duplicated assets will not be committed
** SharedAssets alias is handled during npm build by the setup in react/nwb.config.js

## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


* 

#### @TODO


*

#### Today I learned...
